frontend/src/metabase/lib/urls.js

```diff
@@ -124,18 +124,40 @@ export function tableRowsQuery(databaseId, tableId, metricId, segmentId) {
   return question(null, query);
 }
 
+function slugifyPersonalCollection(collection) {
+  // Current user's personal collection name is replaced with "Your personal collection"
+  // `originalName` keeps the original name like "John Doe's Personal Collection"
+  const name = collection.originalName || collection.name;
+
+  // Will keep single quote characters,
+  // so "John's" can be converted to `john-s` instead of `johns`
+  let slug = slugg(name, {
+    toStrip: /["”“]/g,
+  });
+
+  // If can't build a slug out of user's name (e.g. if it contains only emojis)
+  // removes the "s-" part of a slug
+  if (slug === "s-personal-collection") {
+    slug = slug.replace("s-", "");
+  }
+
+  return slug;
+}
+
 export function collection(collection) {
-  if (
-    !collection ||
-    collection.id === null ||
-    typeof collection.id === "string"
-  ) {
+  const isSystemCollection =
+    !collection || collection.id === null || typeof collection.id === "string";
```

Comment:
We used to reference `/root` collection with just `collection()` when building a URL. I guess changed this, so it's always explicit like `collection({ id: "root" })`. Might be worth to double-check and remove the `!collection` check

Reply:
@alxnddr tried to remove the check, but it makes some tests fail (see [CircleCI report](https://app.circleci.com/pipelines/github/metabase/metabase/16797/workflows/b0533c0a-3115-4859-b75b-3715a6cd19ba/jobs/655964))

Looks like some items can have an `undefined` collection if they belong to the root collection. So if we try to navigate to the collection with a URL like `Urls.collection(props.collection)`, it will fail.

We can also do something like `Urls.collection(props.collection || { id: "root" })`, but I think it will be too noisy as we'll need to update a few different places. So I think having `Urls.collection` fall back to root is acceptable for now

Does it make sense?

Reply:
@kulyk totally, thanks for checking and explaining